register-machines/src/main.rs

```diff
@@ -11,55 +11,203 @@ enum Instruction {
     Halt
 }
 
-use Instruction::*;
-fn eval_program(program: &[Instruction], state: &State) -> State {
-    unimplemented!();
+type Godel = u128;
```

Comment:
Just an itch I have, but Godel could be a trait that u128, usize, BigInt, BigUint, etc could implement. But to be fair you've done so much elsewhere